nexus/db-queries/src/db/pool.rs

```diff
@@ -85,81 +88,60 @@ fn make_postgres_connector(
     ))
 }
 
-impl Pool {
-    /// Creates a new qorb-backed connection pool to the database.
-    ///
-    /// Creating this pool does not necessarily wait for connections to become
-    /// available, as backends may shift over time.
-    pub fn new(log: &Logger, resolver: &QorbResolver) -> Self {
```

Comment:
Since we already have three different constructors for a Pool...

- I didn't want to add an argument to all of them
- I didn't want to add a fourth constructor

So I switched to a builder pattern instead.